### PR TITLE
adapt hypervisor for a1000b and fix the bug of smp startup on a1000

### PR DESCRIPTION
**Pull request by buhenxihuan**

### CODE REVIEW

File: modules/axhal/src/platform/aarch64_bsta1000b/mod.rs [hunk 1]
```diff
@@ -37,6 +37,7 @@ pub(crate) unsafe extern "C" fn rust_entry(cpu_id: usize, dtb: usize) {
 
 #[cfg(feature = "smp")]
 pub(crate) unsafe extern "C" fn rust_entry_secondary(cpu_id: usize) {
+    let cpu_id = cpu_id / 256;
```
Comment (hky1999) [inline modules/axhal/src/platform/aarch64_bsta1000b/mod.rs]: do not hard code this
you can refer to https://github.com/arceos-hypervisor/arceos/blob/c62b386e829cc2252885b90b9d5d7aad9ead6a64/modules/axhal/src/platform/aarch64_bsta1000b/mp.rs#L6C5-L6C11

File: modules/axhal/src/platform/aarch64_common/boot_el2.rs [hunk 2]
```diff
@@ -132,6 +132,9 @@ unsafe extern "C" fn _start() -> ! {
     // PC = 0x8_0000
     // X0 = dtb
     core::arch::asm!("
+        // save DTB pointer
```
Comment (hky1999) [inline modules/axhal/src/platform/aarch64_common/boot_el2.rs:135]: why moving this to the beginning of the code block.

**Review by hky1999 — CHANGES_REQUESTED** [review]

File: modules/axhal/src/platform/aarch64_common/boot_el2.rs (hunk 2 above)
Comment (buhenxihuan) [reply modules/axhal/src/platform/aarch64_common/boot_el2.rs:135]: Prevent the address of dtb from being overwritten by subsequent operations like this：
https://github.com/arceos-hypervisor/arceos/blob/c62b386e829cc2252885b90b9d5d7aad9ead6a64/modules/axhal/src/platform/aarch64_common/boot_el2.rs#L141

File: modules/axhal/src/platform/aarch64_bsta1000b/mod.rs (hunk 1 above)
Comment (buhenxihuan) [reply modules/axhal/src/platform/aarch64_bsta1000b/mod.rs]: yeah, this is appropriate

File: modules/axhal/src/platform/aarch64_common/boot_el2.rs (hunk 2 above)
Comment (hky1999) [reply modules/axhal/src/platform/aarch64_common/boot_el2.rs:135]: get it, thanks

**Review by hky1999 — APPROVED** [review]

**Pull request merged**